### PR TITLE
Adding LuaError on attempt to assign vectors with values out of range

### PR DESCRIPTION
**Pull request by Foghrye4**
Solve issue #4635
A title says its all.
Here is how its looks: screenshot

### CODE REVIEW

**Comment by est31** [comment]
:+1:

File: src/script/common/c_converter.cpp [hunk 1]
```diff
@@ -37,6 +37,14 @@ extern "C" {
 		} \
 	} while(0)
 #define CHECK_POS_COORD(name) CHECK_TYPE(-1, "position coordinate '" name "'", LUA_TNUMBER)
+#define CHECK_FLOAT_RANGE(value, name) \
+if (value < ((float)(s32)((-0x7FFFFFFF - 1) / 1000.0f)) || \
+		value > ((float)(s32)((0x7FFFFFFF) / 1000.0f))) { \
+	char error_text [200]; \
+	sprintf(error_text, "Invalid float vector dimension range '%s' \
+	(expected -2147483.648 < %s < 2147483.647 got %4.3f).",name, name, value); \
```
Comment (sfan5) [inline src/script/common/c_converter.cpp]: These can change based on how specific defines are set (see above)

File: src/script/common/c_converter.cpp [hunk 2]
```diff
@@ -37,6 +37,14 @@ extern "C" {
 		} \
 	} while(0)
 #define CHECK_POS_COORD(name) CHECK_TYPE(-1, "position coordinate '" name "'", LUA_TNUMBER)
+#define CHECK_FLOAT_RANGE(value, name) \
+if (value < ((float)(s32)((-0x7FFFFFFF - 1) / 1000.0f)) || \
+		value > ((float)(s32)((0x7FFFFFFF) / 1000.0f))) { \
```
Comment (sfan5) [inline src/script/common/c_converter.cpp]: Use `F1000_MIN` and `F1000_MAX` from util/serialize.h

File: src/script/common/c_converter.cpp [hunk 3]
```diff
@@ -37,6 +37,14 @@ extern "C" {
...
+	char error_text [200]; \
+	sprintf(error_text, "Invalid float vector dimension range '%s' \
```
Comment (sfan5) [inline src/script/common/c_converter.cpp]: `sprintf` is potentially unsafe, just use a `std::ostringstream`

**Comment by Foghrye4** [comment]
@sfan5 Requested changes are done.

**Comment by nerzhul** [comment]
Okay for me too

**Comment by nerzhul** [comment]
merged, thanks

**Pull request merged**